### PR TITLE
Added thermal throttling detection

### PR DESCRIPTION
**Pull request by ito-san**
IR-004_ブロック図

### CODE REVIEW

**Comment by ito-san** [comment]
IR-004_ブロック図

File: bin/cpu_monitor.py [hunk 1]
```diff
@@ -96,19 +97,19 @@ def __init__(self, hostname, diag_hostname):
...
 
-        self._check_core_temps = rospy.get_param('~check_core_temps', True)
-
         self._cpu_load_warn = rospy.get_param('~cpu_load_warn', cpu_load_warn)
         self._cpu_load_error = rospy.get_param('~cpu_load_error', cpu_load_error)
         self._cpu_load1_warn = rospy.get_param('~cpu_load1_warn', cpu_load1_warn)
         self._cpu_load5_warn = rospy.get_param('~cpu_load5_warn', cpu_load5_warn)
         self._cpu_temp_warn = rospy.get_param('~cpu_temp_warn', cpu_temp_warn)
         self._cpu_temp_error = rospy.get_param('~cpu_temp_error', cpu_temp_error)
+        self._cpu_throttling_error = rospy.get_param('~cpu_throttling_error', cpu_throttling_error)
```
Comment (yn-mrse) [inline bin/cpu_monitor.py:106]: @ito-san Please add this parameter to config/system_monitor.yaml

**Review by yn-mrse — COMMENTED** [review]
@ito-san Have all the additional parts of the code been component tested?

File: bin/cpu_monitor.py (hunk 1 above)
Comment (ito-san) [reply bin/cpu_monitor.py:106]: Addressed in https://github.com/tier4/system_monitor/pull/1/commits/1f835e6617a9563532b3358f16017870a044954f.

**Comment by ito-san** [comment]
> 
> 
> @ito-san Have all the additional parts of the code been component tested?

@yn-mrse 
Yes, please find attached the file. [ThermalThrottling_TestCases.xlsx](https://github.com/tier4/system_monitor/files/4001322/ThermalThrottling_TestCases.xlsx)

**Review by yn-mrse — COMMENTED** [review]
> @yn-mrse
Yes, please find attached the file. ThermalThrottling_TestCases.xlsx

@ito-san 
I checked the attachment.
I think the test content is almost OK.

However, can you confirm the following two points?
1. Testing is performed by creating and destroying test files. Are these files equivalent to the real ones? (the units described inside, or the file permissions etc.)
1. Have you checked whether all conditional branches added this time, operates as expected? (Because this code is not compiled in particular)

**Comment by ito-san** [comment]
@yn-mrse 
>     1. Testing is performed by creating and destroying test files. Are these files equivalent to the real ones? (the units described inside, or the file permissions etc.)

Real sysfs file(e.g. temp1_input) is used for testing as well. When non-zero value is written into test sysfs file(e.g. temp1_test_temp), temp1_input shows temp1_test_temp value instead of real temperature.
**Here is how it works:**
ThermalThrottling_TestCases_Appendix

>     2. Have you checked whether all conditional branches added this time, operates as expected? (Because this code is not compiled in particular)
Added test cases # 6 and 7 to check conditional branching if and catch statement.
Verified as expected.
[ThermalThrottling_TestCases R1.1.xlsx](https://github.com/tier4/system_monitor/files/4002091/ThermalThrottling_TestCases.R1.1.xlsx)

**Pull request merged**